extension/chrome/elements/pgp_block_modules/pgp-block-render-module.ts

```diff
@@ -133,14 +132,12 @@ export class PgpBlockViewRenderModule {
       </body>
       </html>
     `;
-    /* eslint-disable @typescript-eslint/no-non-null-assertion */
-    w!.document.write(html);
+    w?.document.write(html);
     // Give some time for above dom to load in print dialog
     // https://stackoverflow.com/questions/31725373/google-chrome-not-showing-image-in-print-preview
     await Ui.time.sleep(250);
-    w!.window.print();
-    w!.document.close();
-    /* eslint-enable @typescript-eslint/no-non-null-assertion */
+    w?.window.print();
+    w?.document.close();
   };
```

Comment:
I generally recommend going with `!` instead of `?`, against the practice of many developers / teams.

Then things break, I like them to break explicitly. We have a render mechanism for errors that will explain the problem to the user so they can take a screenshot of the problem, and we have an exception logging endpoint to receive information about and fix most common exceptions.

That doesn't necessarily mean that this exact code needs to change back, I think it's fine as you've done it. I'm writing this comment to explain why you see a lot of `!` rather than `?` everywhere, and the rationale behind it.

On iOS, that would have crashed the app. Here, that would throw an exception, which we display, sometimes auto-report, and explain - so the stakes are different.

Reply:
Thanks for explanation 👍 
You're right - after iOS code I've used to avoid `!`